main.py

```
@@ -0,0 +1,195 @@
+    Args:
+        token (str): DNAnexus token_
+    """
+    try:
+        dx_security_context = {"auth_token_type": "Bearer", "auth_token": str(token)}
+
+        dx.set_security_context(dx_security_context)
+        print(dx.api.system_whoami())
+    except dx.exceptions.InvalidAuthentication as err:
+        raise dx.exceptions.InvalidAuthentication(
+            f"DNAnexus Authentication failed: {err}"
+        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)